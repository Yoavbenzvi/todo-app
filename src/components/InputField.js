import React from 'react';

const InputField = ({ addTask, handleInputChange, inputValue }) => {
	return(
		<div className='flex items-center justify-between relative'>
			<input 
				onChange={handleInputChange}
				value={inputValue}
				placeholder='Add a new task...'
				type='text'
				className='p-4 pr-20 border-t-2 border-green-500 rounded bg-gray-900 text-white w-full shadow-inner outline-none"'
			/>
			<button
				onClick={addTask}
				type='button'
				className='text-green-400 hover:text-green-300 bg-gray-900 font-semibold py-2 px-4 absolute right-0 mr-2 focus:outline-none'
			>
				ADD
			</button>
		</div>
	)
}

export default InputField