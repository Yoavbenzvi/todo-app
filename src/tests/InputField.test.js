import { shallow } from "enzyme";
import React from "react";
import InputField from "../components/InputField.js";

describe('InputField component', () => {
	it("expect to render InputField component", () => {

		const wrapper = shallow(<InputField />);
		expect(wrapper).toMatchSnapshot();
	});

	it('testing button click', () => {
		const mockClick = jest.fn();
		const wrapper = shallow(<InputField addTask={mockClick}/>);

		wrapper.find({type: 'button'}).simulate('click');
		expect(mockClick).toHaveBeenCalled();
	})

	it('testing input field change', () => {
		const mockChange = jest.fn();
		const wrapper = shallow(<InputField handleInputChange={mockChange}/>);

		wrapper.find({type: 'text'}).simulate('change')
		expect(mockChange).toHaveBeenCalled();
	})
})
