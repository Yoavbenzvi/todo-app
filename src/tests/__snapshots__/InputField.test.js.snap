// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component expect to render InputField component 1`] = `
<div
  className="flex items-center justify-between relative"
>
  <input
    className="p-4 pr-20 border-t-2 border-green-500 rounded bg-gray-900 text-white w-full shadow-inner outline-none\\""
    placeholder="Add a new task..."
    type="text"
  />
  <button
    className="text-green-400 hover:text-green-300 bg-gray-900 font-semibold py-2 px-4 absolute right-0 mr-2 focus:outline-none"
    type="button"
  >
    ADD
  </button>
</div>
`;
